roofit/roofit/inc/RooLagrangianMorphFunc.h

```diff
@@ -0,0 +1,363 @@
+
+#ifndef ROO_LAGRANGIAN_MORPH
+#define ROO_LAGRANGIAN_MORPH
+
+#include "RooAbsArg.h"
+#include "RooAbsReal.h"
+#include "RooArgList.h"
+#include "RooRatio.h"
+#include "RooRealSumFunc.h"
+#include "RooRealSumPdf.h"
+#include "RooSetProxy.h"
+#include "RooWrapperPdf.h"
+#include "TMatrixD.h"
+
+class RooWorkspace;
+class RooParamHistFunc;
+class RooProduct;
+class RooRealVar;
+class TPair;
+class TFolder;
+class RooLagrangianMorphFunc;
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+class RooLagrangianMorphFunc : public RooAbsReal {
+
+public:
+  typedef std::map<const std::string, double> ParamSet;
+  typedef std::map<const std::string, int> FlagSet;
+  typedef std::map<const std::string, ParamSet> ParamMap;
+  typedef std::map<const std::string, FlagSet> FlagMap;
+
+  class Config {
+  public:
+    Config();
+    Config(const RooAbsCollection &couplings);
+    Config(const RooAbsCollection &prodCouplings,
+           const RooAbsCollection &decCouplings);
+    Config(const Config &other) = default;
+    void setFileName(const char *filename);
+    void setFolders(const RooArgList &folderlist);
+    void setObservableName(const char *obsname);
+    void setCouplings(const RooAbsCollection &couplings);
+    void setCouplings(const RooAbsCollection &prodCouplings,
+                      const RooAbsCollection &decCouplings);
+    void allowNegativeYields(Bool_t allowNegativeYields);
+    template <class T> void setVertices(const std::vector<T> &vertices);
+    template <class T>
+    void setDiagrams(const std::vector<std::vector<T>> &diagrams);
+    template <class T>
+    void setNonInterfering(const std::vector<T *> &nonInterfering);
+    template <class T> void addDiagrams(const std::vector<T> &diagrams);
+
+    std::string getFileName() const;
+    std::string getObservableName() const;
+    std::vector<std::vector<RooArgList *>> getDiagrams() const;
+    RooArgList getCouplings() const;
+    RooArgList getProdCouplings() const;
+    RooArgList getDecCouplings() const;
+    RooArgList getFolders() const;
+    Bool_t IsAllowNegativeYields() const;
+
+    void append(ParamSet &set, const char *str, double val);
+    void append(ParamMap &map, const char *str, ParamSet &set);
+
+    /* WIP
+    void disableInterference(const std::vector<const char*>& nonInterfering) ;
+    void disableInterferences(const std::vector<std::vector<const char*> >&
+    nonInterfering) ;
+    */
+
+    RooRealVar *getParameter(const char *name) const;
+    bool hasParameter(const char *name) const;
+
+    void addFolders(const RooArgList &folders);
+
+    ParamMap const& getParamCards() const { return this->_paramCards; };
+    FlagMap const& getFlagValues() const { return this->_flagValues; };
+
+    std::vector<std::string> const& getFolderNames() const { return _folderNames; };
+    void printSamples() const;
+    void printPhysics() const;
+    int nSamples() const;
+
+void readParameters(TDirectory *f);
```

Comment:
```suggestion
    void readParameters(TDirectory *f);
```
I suggested the wrong indentation earlier. My fault, sorry.